backend/test_observer/data_access/models_enums.py

```diff
@@ -35,6 +35,7 @@ class TestExecutionStatus(str, Enum):
     PASSED = "PASSED"
     FAILED = "FAILED"
     NOT_TESTED = "NOT_TESTED"
+    ENDED = "ENDED"
```

Comment:
Aren't passed and failed tests also ended?

Reply:
Ended here means tests that have completed but do not have test results(eg. provisioning failure).

Reply:
Wouldn't that also be considered failed from the standpoint of this run if provisioning fails?

Reply:
It could be considered a failure, but IIRC, we had a meeting and decided to use a different status/icon for tests that don't have results.

Reply:
Yes it makes sense to have `FAILED` means that there were some tests failing, and `ENDED` meaning that testing didn't complete. Perhaps it's not the best verb to use, as ended doesn't imply that something bad happened. In fact passed or failed both did end. We could call it `ENDED_PREMATURELY`